pkgs/applications/editors/gnome-builder/default.nix

```diff
@@ -0,0 +1,88 @@
+{ stdenv, fetchurl, gnome3, gobjectIntrospection, meson, ninja, pkgconfig
+, appstream-glib, desktop-file-utils, python3, python3Packages, wrapGAppsHook
+, flatpak, gspell, gtk3, gtksourceview3, json-glib, jsonrpc-glib, libdazzle
+, libxml2, ostree, pcre , sysprof, template-glib, vala, webkitgtk
+}:
+let
+  version = "3.28.4";
+  pname = "gnome-builder";
+in stdenv.mkDerivation {
+  name = "${pname}-${version}";
+
+  src = fetchurl {
+    url = "mirror://gnome/sources/${pname}/${gnome3.versionBranch version}/${pname}-${version}.tar.xz";
+    sha256 = "0ibb74jlyrl5f6rj1b74196zfg2qaf870lxgi76qzpkgwq0iya05";
+  };
+
+  nativeBuildInputs = [
+    python3Packages.wrapPython
+    wrapGAppsHook
+    gobjectIntrospection
+    meson
+    ninja
+    pkgconfig
+
+    appstream-glib
+    desktop-file-utils
+  ];
```

Comment:
Unlike https://github.com/NixOS/nixpkgs/pull/33270, this package does not make docs.

Reply:
I hadn't investigated what these were for, I just saw they were optional but not found so I added them. Looks like they aren't about documentation, they're just there to validate that various static files are correctly formatted. You're right, we don't need them at package-build time.

Reply:
I meant that gtk-doc and the docbook deps were missing. appstream-glib is used during build for checking the generated flle: https://gitlab.gnome.org/GNOME/gnome-builder/blob/23e2dc8347d00c2d0d909809b9c965c84ba08752/data/meson.build#L28

Reply:
Oh, I see. But appstream-glib and desktop-file-utils are only used in tests, and with `doCheck` unspecified, we aren't actually running the tests, are we? I mean, maybe we should be, but the other PR didn't either. I'm going to try a build with `doCheck = true`...

Hmm, turns out desktop-file-utils is actually required by `build-aux/meson/post_install.py` as well, so I definitely can't remove that.

Anyway, I'm turning on the documentation building too. I didn't think it was particularly important since it's only documentation for gnome-builder's plugin API, but I guess that's worth having.

Should we have multiple outputs for this package? I can see value in some combination of 'lib', 'dev', and 'devdoc'...

Reply:
> I mean, maybe we should be, but the other PR didn't either.

Yes, if there are tests, we should try to enable them.

> I didn't think it was particularly important since it's only documentation for gnome-builder's plugin API, but I guess that's worth having.

Yup, offline docs are useful for devhelp (if we figure out why it does not work).

> Should we have multiple outputs for this package? I can see value in some combination of 'lib', 'dev', and 'devdoc'...

`out` and `doc`/`devdoc` make sense for apps. `lib` and `dev` only if the package contains a library that is expected to be a dependency of other packages.